xbmc/addons/AddonManager.cpp

```diff
@@ -814,6 +833,15 @@ bool CAddonMgr::IsAddonInstalled(const std::string& ID)
   return GetAddon(ID, tmp, ADDON_UNKNOWN, false);
 }
 
+bool CAddonMgr::IsAddonInstalled(const std::string& ID,
+                                 const std::string& origin,
+                                 const AddonVersion& version)
+{
+  AddonPtr tmp;
+  return (GetAddon(ID, tmp, ADDON_UNKNOWN, false) && tmp->Origin() == origin &&
```

Comment:
What, if `GetAddon`does not return tmp? Can this happen? `tmp->Origin()`would bomb, then....

Reply:
same thing came to my mind. `GetAddon()` will return `false` and `tmp` will contain a nullptr or s.th. else that would bomb. will bring `bool hasAddon` back

Reply:
> If GetAddon returns false.... tmp will not be further accessed, right?

If `GetAddon` is guaranteed to return false if `tmp`is nullptr`then the code is fine as it is.
To be on the safe side I would change the code to:

`return (GetAddon(ID, tmp, ADDON_UNKNOWN, false) && tmp && tmp->Origin() == origin &&`

Reply:
then we're now set with those